check-imgsfx.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+#
+# 2024-01-25 esm
+# 2024-06-28 cevich
+
+set -eo pipefail
+
+# Ensure CWD is the repo root
+cd $(dirname "${BASH_SOURCE[0]}")
+imgsfx=$(<IMG_SFX)
+
+if [[ -z "$HOME" ]]; then
+    echo "FATAL: $$HOME is undefined" >&2
+    exit 1
+fi
+
+# Pre-commit runs in a temp. dir, history must persist via absolute path.
```

Comment:
"runs in a temp dir" -- is that really true? I can't find it in the pre-commit docs. It strikes me as preposterous if true: the entire point of a pre-commit hook is to run tests in the repo, isn't it?

Anyhow, the `cd` on line 9 negates this. You can still keep the history file under `$HOME`, but it's not really a "must". I have a slight preference for keeping it in the same repo dir (possibly with a `.gitignore` entry) but won't block over it.

Reply:
> runs in a temp dir

I couldn't tell if the actual runs happen in a temp. local clone, but certainly the `pre-commit try-repo ...` testing did.  It showed the tmp dir output when I ran it verbosely.  Worse, it would run the script from there as well.  I can try rigging up a `.pre-commit-config.yaml` that points at the local repo. and see if the `pre-push` can use an in-repo dir.

Another thing this comment should probably say:  The `~/.cache/pre-commit` dir. is created/used by the tool itself for logging and whatnot.  So it's not as if I just picked some random location :wink:

Reply:
Comment is now wrong two ways :-)

Reply:
Oh woops did I leave that in there, sorry :relaxed: